### DISCUSSION TITLE
Dev Container Feature to install quarto-cli

### DISCUSSION THREAD

**Discussion by eitsupi** [quarto-dev/quarto-cli#2628] [Show and Tell]
Hello.
I have recently taken to creating a Devcontainer Feature[^1] for to install quarto-cli.
https://github.com/rocker-org/devcontainer-features/tree/main/src/quarto-cli

This may be useful when using quarto-cli on GitHub Codespaces or VSCode Dev Container.

[^1]: https://containers.dev/implementors/features/

**Comment by mcanouil** [comment]
Well done! (also for the RIG Devcontainer feature, <https://github.com/rocker-org/devcontainer-features/blob/main/src/r-rig/README.md>)
I was looking to do this as well since the "custom Devcontainer Feature" feature was released earlier this month[^1].

I'll add this to Awesome Quarto list soon: https://github.com/mcanouil/awesome-quarto/issues/124

[^1]: https://code.visualstudio.com/blogs/2022/09/15/dev-container-features

**Comment by agerlach** [comment]
This is awesome and extremely helpful. Is there any plans on adding an `installChromium` option?

https://quarto.org/docs/authoring/diagrams.html#chrome-install

**Reply by eitsupi** [reply]
> This is awesome and extremely helpful. Is there any plans on adding an `installChromium` option?

No.
As noted on the website, `quarto tools install chromium` does not work on Docker containers.
Users must install Chromium or Google Chrome on their own.

**Reply by agerlach** [reply]
makes sense. Thanks

**Reply by eitsupi** [reply]
Hi, I added a new option to combine with the `apt-packages` Feature to install Chromium.
The following devcontainer.json should work.

```json
{
    "image": "mcr.microsoft.com/devcontainers/base:debian",
    "features": {
        "ghcr.io/rocker-org/devcontainer-features/quarto-cli:1": {
            "installChromium": true
        },
        "ghcr.io/rocker-org/devcontainer-features/apt-packages:1": {
            "packages": "chromium"
        }
    }
}
```

Please check <https://github.com/rocker-org/devcontainer-features/tree/main/src/quarto-cli>
